### DISCUSSION TITLE
Legend-State automatically tracks observables, is this possible in MobX?

### DISCUSSION THREAD

**Discussion by Obiwarn** [mobxjs/mobx#3843] [Ideas]
I just took a look at legend-state: https://legendapp.com/open-source/state/intro/introduction/

It seems that it automatically tracks its observables. Are there any drawbacks to this? Is there a reason this is not possible with MobX?

```
// Enable React components to automatically track observables
enableReactTracking({ auto: true });

const Component = function Component() {
  // get() makes this component re-render whenever theme changes
  const theme = state$.settings.theme.get();

  return <div>Theme: {theme}</div>;
};
```

**Comment by kubk** [comment]
It has been already discussed here: https://github.com/mobxjs/mobx/discussions/3528

The author's conclusion: https://github.com/mobxjs/mobx/discussions/3528#discussioncomment-3813489

I'd recommend creating IDE snippet to create a component already wrapped with observer.

**Reply by Obiwarn** [reply]
Thank you very much for the links.

Sad news.
